index.js

```diff
@@ -38,25 +38,37 @@ const WinstonDynamoDB = function (options) {
 
 util.inherits(WinstonDynamoDB, winston.Transport);
 
+
+
 WinstonDynamoDB.prototype.log = function (info, callback) {
-    debug('log (called by winston)', info);
+    const  internalLog = (info, callback) => {
+        debug('log (called by winston)', info);
 
-    if (!isEmpty(info.message) || isError(info.message)) {
-        this.add(info);
-    }
+        if (!isEmpty(info.message) || isError(info.message)) {
+            this.add(info);
+        }
+
+        if (!/^uncaughtException: /.test(info.message)) {
+            // do not wait, just return right away
+            return callback(null, true);
+        }
 
-    if (!/^uncaughtException: /.test(info.message)) {
-        // do not wait, just return right away
-        return callback(null, true);
+        debug('message not empty, proceeding')
+
+        // clear interval and send logs immediately
+        // as Winston is about to end the process
+        clearInterval(this.intervalId);
+        this.intervalId = null;
+        this.submit(callback);
     }
 
-    debug('message not empty, proceeding')
+    const { message } = info;
 
-    // clear interval and send logs immediately
-    // as Winston is about to end the process
-    clearInterval(this.intervalId);
-    this.intervalId = null;
-    this.submit(callback);
+    for (let i = 0; i < message.length; i+=10_000) {
+        internalLog({ ...info, message: message.slice(i, i+10_000) }, () => {});
+    }
+
+    callback();
```

Comment:
So there are a couple of things here that should be adapted

1. `uncaughtException` logic is for [a configuration that allows logging exceptions in winston](https://github.com/winstonjs/winston?tab=readme-ov-file#handling-uncaught-exceptions-with-winston). In such a case we should just add and then clear the interval, no need to split up the events (we can assume an internal error in winston will not be that long).
We should probably fix that, it will also make the code more readable here
2. In the happy path we want to send `callback(null, true)` and not `callback()`.
3. We only want to perform this splitting logic if `options.jsonMessage`, and we need to trim the `message.message` only, and not the `message` which is a stringified JSON in itself
4. We should try to have the message as large as possible, to avoid having the logging take too long by sending out multiple batches to DDB. So we should raise from 10,000 to 300,000 (which is probably good enough to assume that all metadata in the `message` JSON is not gonna take up 100,000 bytes)